### ISSUE TITLE
Archived threads aren't marked as read, so it's hard to follow progress on a thread

### ISSUE THREAD

**Issue by Ecnassianer** [gornostal/owa-enhancements#1]
It would be great if threads that were archived were also marked as read. That way when you go back to a new email in a thread, you'll see all the ones you've already read. 

I'm pretty unfamiliar with Javascript, but I tried adding this function to the following function to the script. It opens the actions menu, but doesn't actually mark the thread as read:

```
function markAsReadCurrentMessage() {
    var actionBtn = $('#divActions'),
        markAsReadBtnSelector = "#divCMM #spnT:contains('Mark As Read')";

    if(!actionBtn.length){
        return;
    }
    actionBtn.click();
    waitForSubmenu().done(function(archiveBtn){
        archiveBtn.click();
    });

    /**
     * @return {$.Deferred}
     * Shitty copy paste code above cause I don't know javascript enough to extract out a deferred-safe function
     */
    function waitForSubmenu(){
        var onSubmenu = new $.Deferred(),
            interval = 50;

        e(function checkSubmenu (){
            var archiveBtn = $(markAsReadBtnSelector);
            if(archiveBtn.length) {
                onSubmenu.resolveWith(this, [archiveBtn]);
            } else {
                setTimeout(checkSubmenu, interval);
            }
        }, interval);

        return onSubmenu;
    }
}
```

**Comment by gornostal** [comment]
Hi @Ecnassianer 

It's good to know that there are people who find my script useful.
But, I'm no longer using it myself, therefore I'm not going to support it or add new features.

From your code it's unclear what may be wrong. I would need to debug to find that out.
Sorry I could not help you.
